ci/validate-version-bump.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+# This script checks if a crate needs a version bump.
+#
+# At the time of writing, it doesn't check what kind of bump is required.
+# In the future, we could take SemVer compatibliity into account, like
+# integrating `cargo-semver-checks` of else
+#
+# Inputs:
+#     BASE_SHA    The commit SHA of the branch where the PR wants to merge into.
+#     COMMIT_SHA  The commit SHA that triggered the workflow.
+
+set -euo pipefail
+
+# When `BASE_SHA` is missing, we assume it is from bors merge commit,
+# so `HEAD~` should find the previous commit on master branch.
+base_sha=$(git rev-parse "${BASE_SHA:-HEAD~}")
+commit_sha=$(git rev-parse "${COMMIT_SHA:-HEAD}")
```

Comment:
I was assuming this would be written as an xtask, rather than a shell script wrapping one.  It'd make it easier to identify changes that way.